core/pom.xml

```diff
@@ -518,7 +518,7 @@ THE SOFTWARE.
     <dependency> <!-- Not included into BOM, plugins should use one from structs-plugin -->
       <groupId>org.jenkins-ci</groupId>
       <artifactId>symbol-annotation</artifactId>
-      <version>1.21</version>
+      <version>1.1</version>
```

Comment:
why not just bump the version of structs? you just need to use bom version 24.

I noticed this in https://github.com/jenkinsci/plot-plugin/pull/65, but bumping bom fixed it

Reply:
The particular version does not matter; it has not changed since 1.0 AFAIK. I just do not want it to be constantly bumped up here.

Reply:
Which one is actually used at runtime? the structs one or the Jenkins core one?

Reply:
The Jenkins core one. Which does not really matter because

```bash
git diff structs-parent-1.1 -- annotation/src/main
```

```diff
diff --git annotation/src/main/java/org/jenkinsci/Symbol.java annotation/src/main/java/org/jenkinsci/Symbol.java
index abd45ef..7f007e2 100644
--- annotation/src/main/java/org/jenkinsci/Symbol.java
+++ annotation/src/main/java/org/jenkinsci/Symbol.java
@@ -33,6 +33,10 @@ import static java.lang.annotation.RetentionPolicy.*;
  * The first one is used as the primary identifier for reverse-mapping.
  *
  * <p>
+ * Generally symbols should be named after their display names, so that users see
+ * association between what they see in web UI and symbols they see in code.
+ *
+ * <p>
  * To look up a component by its symbol, see the documentation of the symbol plugin.
  */
 @Indexed
```

Reply:
The last code change was in 1.0, so it does not matter much: https://github.com/jenkinsci/structs-plugin/commit/2efc80dc826685a43f717595d64307ad7f474a44#diff-94e7f2d13ab308fa2df75a0097ab64be6d274e2a8233741ca4c45f1195602c71 . The produced bytecode is likely to be different between versions though.

Reply:
I still don't understand why structs doesn't just pickup the jenkins core one?

Shouldn't we just remove the dependency there, what if we did want to update the library?

Reply:
> why structs doesn't just pickup the jenkins core one

It could, if the annotation component were split to its own repo with its own lifecycle.

Reply:
> It could, if the annotation component were split to its own repo with its own lifecycle.

@timja @jglick I extracted this to a separate repository here: https://github.com/basil/lib-symbol-annotation

What are the next steps? Get this hosted under the `jenkinsci` GitHub organization, add myself to RPU and cut a release, and then update core to use the new release and `structs` plugin to use the version from core? Can the former be sorted casually or shall I file a `HOSTING` request?

Reply:
Can do you a hosting request please.

Reply:
Done: [HOSTING-1155](https://issues.jenkins.io/browse/HOSTING-1155)